src/jit/importer.cpp

```diff
@@ -3895,14 +3922,27 @@ NamedIntrinsic Compiler::lookupNamedIntrinsic(CORINFO_METHOD_HANDLE method)
     const char* namespaceName = nullptr;
     const char* methodName    = info.compCompHnd->getMethodNameFromMetadata(method, &className, &namespaceName);
 
-    if ((namespaceName != nullptr) && strcmp(namespaceName, "System") == 0)
+    // Currently we only have intrinsics at the method level, so we can check that
+    // namespaceName, className, and methodName are all not null upfront. If this ever
+    // changes (such as if we support all methods in a class as intrinsic), we can reorder
+    // the checks to account for that.
+
+    if ((namespaceName != nullptr) && (className != nullptr) && (methodName != nullptr))
```

Comment:
I'm fine with the reordering to put all the null checks up front. I have something similar in some of my pending changes. 

However, I believe the jit will always be asking if a particular *method* is an intrinsic here. So I suggest the comment be removed or reworded.

Reply:
Fixed to no longer include the part about reordering if it ever changes.